### PR TITLE
Add a Detekt rule to enforce empty lines after blocks 

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: detekt-rules/src/main/kotlin/OrtEmptyLineAfterBlock.kt [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import org.jetbrains.kotlin.psi.KtBlockExpression
+import org.jetbrains.kotlin.psi.KtExpression
+import org.jetbrains.kotlin.psi.KtLambdaExpression
+import org.jetbrains.kotlin.psi.psiUtil.allChildren
+
+class OrtEmptyLineAfterBlock(config: Config) : Rule(config) {
+    override val issue = Issue(
+        javaClass.simpleName,
+        Severity.Style,
+        "Reports files that do not follow ORT's convention of having empty lines after blocks",
```
Comment (fviernau) [inline detekt-rules/src/main/kotlin/OrtEmptyLineAfterBlock.kt:43]: nit: The convention bit could be ommited because it seems redundant.

Comment (fviernau) [reply detekt-rules/src/main/kotlin/OrtEmptyLineAfterBlock.kt:43]: Maybe: "Reports missing empty lines following any code block.

File: detekt-rules/src/main/kotlin/OrtEmptyLineAfterBlock.kt [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        checkExpression(blockExpression)
+    }
+
+    override fun visitLambdaExpression(lambdaExpression: KtLambdaExpression) {
+        super.visitLambdaExpression(lambdaExpression)
+        checkExpression(lambdaExpression)
+    }
+
+    private fun checkExpression(expression: KtExpression) {
+        // Only care about blocks than span multiple lines.
```
Comment (fviernau) [inline detekt-rules/src/main/kotlin/OrtEmptyLineAfterBlock.kt:58]: tahn -> that

File: detekt-rules/src/main/kotlin/OrtEmptyLineAfterBlock.kt [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        if (!expression.hasNewLine()) return
+
+        // Find the next expression after the block, if any.
+        var currentElement: PsiElement = expression
+        while (currentElement.nextSibling == null) {
+            currentElement = currentElement.parent ?: return
+        }
+
+        val firstElementAfterBlock = currentElement.nextSibling ?: return
+        if (!firstElementAfterBlock.isNewLine()) return
```
Comment (fviernau) [inline detekt-rules/src/main/kotlin/OrtEmptyLineAfterBlock.kt:68]: nit: Maybe use `takeIf .. ?: return`

File: detekt-rules/src/main/kotlin/OrtEmptyLineAfterBlock.kt [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+                issue,
+                // Use the message as the name to also see it in CLI output and not only in the report files.
+                Entity.from(expression).copy(name = message),
+                message
+            )
+
+            report(finding)
+        }
+    }
+}
```
Comment (fviernau) [inline detekt-rules/src/main/kotlin/OrtEmptyLineAfterBlock.kt:86]: IIUC this does return just problematic files. Would it make sense to instead return file locations, e.g. `path, line#)` instead?

**Review by fviernau — CHANGES_REQUESTED** [review]

File: detekt-rules/src/main/kotlin/OrtEmptyLineAfterBlock.kt (hunk 4 above)
Comment (sschuberth) [reply detekt-rules/src/main/kotlin/OrtEmptyLineAfterBlock.kt:86]: The path and line number (of the start of the block that lacks the following empty line) is information that's printed by Detekt automatically.

File: detekt-rules/src/main/kotlin/OrtEmptyLineAfterBlock.kt (hunk 3 above)
Comment (sschuberth) [reply detekt-rules/src/main/kotlin/OrtEmptyLineAfterBlock.kt:68]: I'd prefer to stick to `if` here to resemble the formatting for the following `secondElementAfterBlock` logic a bit.

**Review by fviernau — APPROVED** [review]

**Pull request merged**